<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>660</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Flood Fill</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>6</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>6</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QScrollArea" name="scrollArea">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <widget class="QcvMatWidget" name="widgetImage">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>626</width>
         <height>509</height>
        </rect>
       </property>
      </widget>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>150</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>150</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="captureTab">
       <attribute name="title">
        <string>Size</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="leftMargin">
         <number>8</number>
        </property>
        <property name="topMargin">
         <number>8</number>
        </property>
        <property name="rightMargin">
         <number>8</number>
        </property>
        <property name="bottomMargin">
         <number>8</number>
        </property>
        <item>
         <widget class="QRadioButton" name="radioButtonOrigSize">
          <property name="text">
           <string>Originale</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButtonCustomSize">
          <property name="text">
           <string>Custom</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelWidth">
          <property name="text">
           <string>Largeur</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinBoxWidth">
          <property name="maximum">
           <number>1600</number>
          </property>
          <property name="singleStep">
           <number>4</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelHeight">
          <property name="text">
           <string>Hauteur</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinBoxHeight">
          <property name="maximum">
           <number>1200</number>
          </property>
          <property name="singleStep">
           <number>4</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBoxFlip">
          <property name="text">
           <string>Flip</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBoxGray">
          <property name="text">
           <string>Gray</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>228</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="processTab">
       <attribute name="title">
        <string>Process</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="leftMargin">
         <number>8</number>
        </property>
        <property name="topMargin">
         <number>8</number>
        </property>
        <property name="rightMargin">
         <number>8</number>
        </property>
        <property name="bottomMargin">
         <number>8</number>
        </property>
        <item>
         <widget class="QGroupBox" name="groupBoxDisplay">
          <property name="title">
           <string>Display</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <property name="leftMargin">
            <number>8</number>
           </property>
           <property name="topMargin">
            <number>8</number>
           </property>
           <property name="rightMargin">
            <number>8</number>
           </property>
           <property name="bottomMargin">
            <number>8</number>
           </property>
           <item>
            <widget class="QRadioButton" name="radioButtonInput">
             <property name="text">
              <string>Input</string>
             </property>
             <property name="shortcut">
              <string>I</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButtonMask">
             <property name="text">
              <string>Mask</string>
             </property>
             <property name="shortcut">
              <string>M</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButtonMerged">
             <property name="text">
              <string>Merged</string>
             </property>
             <property name="shortcut">
              <string>E</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxShow">
          <property name="title">
           <string>Show</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_9">
           <property name="leftMargin">
            <number>2</number>
           </property>
           <property name="topMargin">
            <number>2</number>
           </property>
           <property name="rightMargin">
            <number>2</number>
           </property>
           <property name="bottomMargin">
            <number>2</number>
           </property>
           <item>
            <widget class="QCheckBox" name="checkBoxBBox">
             <property name="text">
              <string>Bbox</string>
             </property>
             <property name="shortcut">
              <string>B</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBoxCenter">
             <property name="text">
              <string>Center</string>
             </property>
             <property name="shortcut">
              <string>S</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxConnectivity">
          <property name="title">
           <string>Connectivity</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <property name="leftMargin">
            <number>2</number>
           </property>
           <property name="topMargin">
            <number>2</number>
           </property>
           <property name="rightMargin">
            <number>2</number>
           </property>
           <property name="bottomMargin">
            <number>2</number>
           </property>
           <item>
            <widget class="QRadioButton" name="radioButton4Connect">
             <property name="text">
              <string>4 Pixels</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton8Connect">
             <property name="text">
              <string>8 Pixels</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxFloodFill">
          <property name="title">
           <string>Flood Threshold</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <property name="leftMargin">
            <number>2</number>
           </property>
           <property name="topMargin">
            <number>2</number>
           </property>
           <property name="rightMargin">
            <number>2</number>
           </property>
           <property name="bottomMargin">
            <number>2</number>
           </property>
           <item>
            <widget class="QRadioButton" name="radioButtonAbsThreshold">
             <property name="text">
              <string>Absolute</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButtonRelThreshold">
             <property name="text">
              <string>Relative</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonClearFlood">
          <property name="text">
           <string>Clear</string>
          </property>
          <property name="shortcut">
           <string>X</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonNewColor">
          <property name="text">
           <string>New Color</string>
          </property>
          <property name="shortcut">
           <string>N, Space</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="labelTTime">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Time</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelTime">
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>0</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="1" column="0" colspan="2">
     <widget class="QGroupBox" name="groupBoxThreshold">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Flood Thresholds</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="leftMargin">
        <number>8</number>
       </property>
       <property name="rightMargin">
        <number>8</number>
       </property>
       <property name="bottomMargin">
        <number>8</number>
       </property>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <widget class="QLabel" name="labelLoDiff">
           <property name="text">
            <string>Low difference</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="labelUpDiff">
           <property name="text">
            <string>Up difference</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_7">
         <item>
          <widget class="QSlider" name="horizontalSliderLoDiff">
           <property name="maximum">
            <number>255</number>
           </property>
           <property name="value">
            <number>20</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="horizontalSliderUpDiff">
           <property name="maximum">
            <number>255</number>
           </property>
           <property name="value">
            <number>20</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_8">
         <item>
          <widget class="QSpinBox" name="spinBoxLoDiff">
           <property name="maximum">
            <number>255</number>
           </property>
           <property name="value">
            <number>20</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBoxUpDiff">
           <property name="maximum">
            <number>255</number>
           </property>
           <property name="value">
            <number>20</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBoxLink">
         <property name="text">
          <string>Link</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuSources">
    <property name="title">
     <string>Sources</string>
    </property>
    <addaction name="actionCamera_0"/>
    <addaction name="actionCamera_1"/>
    <addaction name="actionFile"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuVideo">
    <property name="title">
     <string>Video</string>
    </property>
    <widget class="QMenu" name="menuSize">
     <property name="title">
      <string>taille</string>
     </property>
     <addaction name="actionOriginalSize"/>
     <addaction name="actionConstrainedSize"/>
    </widget>
    <addaction name="actionFlip"/>
    <addaction name="actionGray"/>
    <addaction name="separator"/>
    <addaction name="menuSize"/>
   </widget>
   <widget class="QMenu" name="menuRender">
    <property name="title">
     <string>Render</string>
    </property>
    <addaction name="actionRenderImage"/>
    <addaction name="actionRenderPixmap"/>
    <addaction name="actionRenderOpenGL"/>
   </widget>
   <addaction name="menuSources"/>
   <addaction name="menuVideo"/>
   <addaction name="menuRender"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionCamera_0">
   <property name="text">
    <string>Camera 0</string>
   </property>
   <property name="toolTip">
    <string>caméra interne</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+0</string>
   </property>
  </action>
  <action name="actionCamera_1">
   <property name="text">
    <string>Camera 1</string>
   </property>
   <property name="toolTip">
    <string>caméra externe</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+1</string>
   </property>
  </action>
  <action name="actionFile">
   <property name="text">
    <string>File</string>
   </property>
   <property name="toolTip">
    <string>fichier video</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionFlip">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>flip</string>
   </property>
  </action>
  <action name="actionOriginalSize">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Originale</string>
   </property>
   <property name="toolTip">
    <string>taille originale</string>
   </property>
  </action>
  <action name="actionConstrainedSize">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Custom</string>
   </property>
   <property name="iconText">
    <string>contrainte</string>
   </property>
   <property name="toolTip">
    <string>taille imposée</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quitter</string>
   </property>
  </action>
  <action name="actionRenderImage">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Image</string>
   </property>
  </action>
  <action name="actionRenderPixmap">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pixmap</string>
   </property>
  </action>
  <action name="actionRenderOpenGL">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>OpenGL</string>
   </property>
  </action>
  <action name="actionGray">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>gray</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QcvMatWidget</class>
   <extends>QWidget</extends>
   <header>QcvMatWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>scrollArea</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>radioButtonCustomSize</sender>
   <signal>clicked(bool)</signal>
   <receiver>actionConstrainedSize</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>631</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionConstrainedSize</sender>
   <signal>triggered(bool)</signal>
   <receiver>radioButtonCustomSize</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>631</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButtonOrigSize</sender>
   <signal>clicked(bool)</signal>
   <receiver>actionOriginalSize</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>597</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOriginalSize</sender>
   <signal>triggered(bool)</signal>
   <receiver>radioButtonOrigSize</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>597</x>
     <y>54</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxFlip</sender>
   <signal>clicked(bool)</signal>
   <receiver>actionFlip</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>640</x>
     <y>233</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFlip</sender>
   <signal>triggered(bool)</signal>
   <receiver>checkBoxFlip</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>640</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxGray</sender>
   <signal>clicked(bool)</signal>
   <receiver>actionGray</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>576</x>
     <y>244</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionGray</sender>
   <signal>triggered(bool)</signal>
   <receiver>checkBoxGray</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>576</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBoxLoDiff</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSliderLoDiff</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>588</x>
     <y>476</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>480</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSliderLoDiff</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBoxLoDiff</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>467</x>
     <y>477</y>
    </hint>
    <hint type="destinationlabel">
     <x>595</x>
     <y>474</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBoxUpDiff</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSliderUpDiff</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>602</x>
     <y>515</y>
    </hint>
    <hint type="destinationlabel">
     <x>540</x>
     <y>513</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSliderUpDiff</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBoxUpDiff</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>463</x>
     <y>514</y>
    </hint>
    <hint type="destinationlabel">
     <x>602</x>
     <y>512</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
